stdlib/public/core/SequenceAlgorithms.swift.gyb

```diff
@@ -620,6 +620,64 @@ extension Sequence {
   }
 }
 
+
+//===----------------------------------------------------------------------===//
+// reduce(into:_:)
+//===----------------------------------------------------------------------===//
+
+extension Sequence {
+  /// Returns the result of combining the elements of the sequence using the
+  /// given closure.
+  ///
+  /// Use the `reduce(into:_:)` method to produce a single value from the elements
+  /// of an entire sequence. For example, you can use this method on an array
+  /// of numbers to find their sum or product.
+  ///
+  /// The `nextPartialResult` closure is called sequentially with an
+  /// accumulating value initialized to `initialResult` and each element of
+  /// the sequence. This example shows how to find the sum of an array of
+  /// numbers.
```

Comment:
Would a different example help make it clear when to use this version of reduce instead of the other? Maybe something like the `uniq` method from the proposal:

```swift
let numbers = [1, 2, 2, 3, 3, 3, 4, 4, 4, 4]
let noSequentials = numbers.reduce(into: [] as [Int]) { array, element in
    if element != array.last { 
        array.append(element) 
    }
}
// noSequentials == [1, 2, 3, 4]
```